driver/index.js

```diff
@@ -0,0 +1,5 @@
+const { handlePickup, handleDelivered } = require('./handler');
+const eventPool = require('../eventPool');
+
+eventPool.on('pickup', handlePickup);
+eventPool.on('delivered', handleDelivered);
```

Comment:
You've got a loop here -- a 'delivered' event triggers your handler which ends by emitting a 'delivered' event.